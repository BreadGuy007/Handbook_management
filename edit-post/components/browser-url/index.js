/**
 * WordPress dependencies
 */
import { Component } from '@wordpress/element';
import { withSelect } from '@wordpress/data';
import { addQueryArgs } from '@wordpress/url';

/**
 * Returns the Post's Edit URL.
 *
 * @param {number} postId Post ID.
 *
 * @return {string} Post edit URL.
 */
export function getPostEditURL( postId ) {
	return addQueryArgs( 'post.php', { post: postId, action: 'edit' } );
}

export class BrowserURL extends Component {
	constructor() {
		super( ...arguments );

		this.state = {
			historyId: null,
		};
	}

	componentDidUpdate( prevProps ) {
		const { postId, postStatus } = this.props;
		const { historyId } = this.state;
		if ( postId === prevProps.postId && postId === historyId ) {
			return;
		}

		if ( postStatus !== 'auto-draft' ) {
			this.setBrowserURL( postId );
		}
	}

	/**
	 * Replaces the browser URL with a post editor link for the given post ID.
	 *
	 * Note it is important that, since this function may be called when the
	 * editor first loads, the result generated `getPostEditURL` matches that
	 * produced by the server. Otherwise, the URL will change unexpectedly.
	 *
	 * @param {number} postId Post ID for which to generate post editor URL.
	 */
	setBrowserURL( postId ) {
		window.history.replaceState(
			{ id: postId },
			'Post ' + postId,
			getPostEditURL( postId )
		);

		this.setState( () => ( {
			historyId: postId,
		} ) );
	}

	render() {
		return null;
	}
}

export default withSelect( ( select ) => {
	const { getCurrentPost } = select( 'core/editor' );
	const { id, status } = getCurrentPost();

	return {
		postId: id,
		postStatus: status,
	};
} )( BrowserURL );
