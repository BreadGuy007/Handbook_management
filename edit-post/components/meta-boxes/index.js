/**
 * WordPress dependencies
 */
import { withSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import MetaBoxesArea from './meta-boxes-area';
import MetaBoxesPanel from './meta-boxes-panel';

function MetaBoxes( { location, isActive, usePanel = false } ) {
	if ( ! isActive ) {
		return null;
	}

	const element = <MetaBoxesArea location={ location } />;

	if ( ! usePanel ) {
		return element;
	}

	return (
		<MetaBoxesPanel>
			{ element }
		</MetaBoxesPanel>
	);
}

export default withSelect(
	( select, ownProps ) => ( {
		isActive: select( 'core/edit-post' ).getMetaBox( ownProps.location ).isActive,
	} ),
)( MetaBoxes );
