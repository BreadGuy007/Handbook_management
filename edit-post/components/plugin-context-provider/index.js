/**
 * External dependencies
 */
import { noop } from 'lodash';

/**
 * WordPress dependencies
 */
import { Component } from '@wordpress/element';

/**
 * Provives context to plugins, in combination with the withPluginContext
 * higher order component.
 */
class PluginContextProvider extends Component {
	getChildContext() {
		return {
			pluginContext: this.props.value,
		};
	}

	render() {
		return this.props.children;
	}
}

PluginContextProvider.childContextTypes = {
	pluginContext: noop,
};

export default PluginContextProvider;
