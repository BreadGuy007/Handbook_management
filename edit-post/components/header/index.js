/**
 * External dependencies
 */
import { connect } from 'react-redux';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { IconButton } from '@wordpress/components';
import {
	PostPreviewButton,
	PostSavedState,
	PostPublishPanelToggle,
} from '@wordpress/editor';

/**
 * Internal dependencies
 */
import './style.scss';
import MoreMenu from './more-menu';
import HeaderToolbar from './header-toolbar';
import {
	getOpenedGeneralSidebar,
	isPublishSidebarOpened,
	hasMetaBoxes,
	isSavingMetaBoxes,
} from '../../store/selectors';
import {
	openGeneralSidebar,
	closeGeneralSidebar,
	togglePublishSidebar,
} from '../../store/actions';

function Header( {
	isGeneralSidebarEditorOpen,
	onOpenGeneralSidebar,
	onCloseGeneralSidebar,
	isPublishSidebarOpen,
	onTogglePublishSidebar,
	hasActiveMetaboxes,
	isSaving,
} ) {
	const toggleGeneralSidebar = isGeneralSidebarEditorOpen ? onCloseGeneralSidebar : onOpenGeneralSidebar;

	return (
		<div
			role="region"
			aria-label={ __( 'Editor toolbar' ) }
			className="edit-post-header"
			tabIndex="-1"
		>
			<HeaderToolbar />
			{ ! isPublishSidebarOpen && (
				<div className="edit-post-header__settings">
					<PostSavedState
						forceIsDirty={ hasActiveMetaboxes }
						forceIsSaving={ isSaving }
					/>
					<PostPreviewButton />
					<PostPublishPanelToggle
						isOpen={ isPublishSidebarOpen }
						onToggle={ onTogglePublishSidebar }
					/>
					<IconButton
						icon="admin-generic"
						onClick={ toggleGeneralSidebar }
						isToggled={ isGeneralSidebarEditorOpen }
						label={ __( 'Settings' ) }
						aria-expanded={ isGeneralSidebarEditorOpen }
					/>
					<MoreMenu key="more-menu" />
				</div>
			) }
		</div>
	);
}

export default connect(
	( state ) => ( {
		isGeneralSidebarEditorOpen: getOpenedGeneralSidebar( state ) === 'editor',
		isPublishSidebarOpen: isPublishSidebarOpened( state ),
		hasActiveMetaboxes: hasMetaBoxes( state ),
		isSaving: isSavingMetaBoxes( state ),
	} ),
	{
		onOpenGeneralSidebar: () => openGeneralSidebar( 'editor' ),
		onCloseGeneralSidebar: closeGeneralSidebar,
		onTogglePublishSidebar: togglePublishSidebar,
	},
	undefined,
	{ storeKey: 'edit-post' }
)( Header );
