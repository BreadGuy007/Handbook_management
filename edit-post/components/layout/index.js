/**
 * External dependencies
 */
import { connect } from 'react-redux';
import classnames from 'classnames';

/**
 * WordPress dependencies
 */
import { Popover, navigateRegions } from '@wordpress/components';
import { __ } from '@wordpress/i18n';
import {
	MetaBoxes,
	AutosaveMonitor,
	UnsavedChangesWarning,
	EditorNotices,
	PostPublishPanel,
	DocumentTitle,
} from '@wordpress/editor';

/**
 * Internal dependencies
 */
import './style.scss';
import Header from '../header';
import Sidebar from '../sidebar';
import TextEditor from '../modes/text-editor';
import VisualEditor from '../modes/visual-editor';
import EditorModeKeyboardShortcuts from '../modes/keyboard-shortcuts';
import {
	getEditorMode,
	hasFixedToolbar,
	hasOpenSidebar,
	isSidebarOpened,
} from '../../store/selectors';
import { toggleSidebar } from '../../store/actions';

function Layout( {
	mode,
	layoutHasOpenSidebar,
	isDefaultSidebarOpened,
	isPublishSidebarOpened,
	fixedToolbarActive,
	onClosePublishPanel,
} ) {
	const className = classnames( 'edit-post-layout', {
		'is-sidebar-opened': layoutHasOpenSidebar,
		'has-fixed-toolbar': fixedToolbarActive,
	} );

	return (
		<div className={ className }>
			<DocumentTitle />
			<UnsavedChangesWarning />
			<AutosaveMonitor />
			<Header />
			<div className="edit-post-layout__content" role="region" aria-label={ __( 'Editor content' ) } tabIndex="-1">
				<EditorNotices />
				<div className="edit-post-layout__editor">
					<EditorModeKeyboardShortcuts />
					{ mode === 'text' && <TextEditor /> }
					{ mode === 'visual' && <VisualEditor /> }
				</div>
				<div className="edit-post-layout__metaboxes">
					<MetaBoxes location="normal" />
				</div>
				<div className="edit-post-layout__metaboxes">
					<MetaBoxes location="advanced" />
				</div>
			</div>
			{ isDefaultSidebarOpened && <Sidebar /> }
			{ isPublishSidebarOpened && <PostPublishPanel onClose={ onClosePublishPanel } /> }
			<Popover.Slot />
		</div>
	);
}

export default connect(
	( state ) => ( {
		mode: getEditorMode( state ),
		layoutHasOpenSidebar: hasOpenSidebar( state ),
		isDefaultSidebarOpened: isSidebarOpened( state ),
		isPublishSidebarOpened: isSidebarOpened( state, 'publish' ),
		fixedToolbarActive: hasFixedToolbar( state ),
	} ),
	{
		onClosePublishPanel: () => toggleSidebar( 'publish', false ),
	},
	undefined,
	{ storeKey: 'edit-post' }
)( navigateRegions( Layout ) );
