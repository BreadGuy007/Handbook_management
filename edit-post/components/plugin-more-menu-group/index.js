/**
 * WordPress dependencies
 */
import { compose } from '@wordpress/element';
import { withContext, MenuGroup } from '@wordpress/components';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import PluginMoreMenuItem, { SLOT_NAME } from '../plugin-more-menu-item';

const PluginMoreMenuGroup = ( { getFills, fillProps } ) => {
	// We don't want the plugins menu items group to be rendered if there are no fills.
	if ( ! getFills( SLOT_NAME ).length ) {
		return null;
	}
	return (
		<MenuGroup
			label={ __( 'Plugins' ) } >
			<PluginMoreMenuItem.Slot name={ SLOT_NAME } fillProps={ fillProps } />
		</MenuGroup>
	);
};

export default compose( [
	withContext( 'getFills' )(),
] )( PluginMoreMenuGroup );
