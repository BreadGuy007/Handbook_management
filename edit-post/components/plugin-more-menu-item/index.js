/**
 * External dependencies
 */
import { flow, noop } from 'lodash';

/**
 * WordPress dependencies
 */
import { Slot, Fill } from '@wordpress/components';
import { PluginContext } from '@wordpress/plugins';

/**
 * Internal dependencies
 */
import PluginSidebarMoreMenuItem from './plugin-sidebar-more-menu-item';

/**
 * Name of slot in which the more menu items should fill.
 *
 * @type {string}
 */
export const SLOT_NAME = 'PluginMoreMenuItem';

const PluginMoreMenuItem = ( props ) => (
	<PluginContext.Consumer>
		{ ( { pluginName } ) => (
			<Fill name={ SLOT_NAME }>
				{ ( fillProps ) => {
					const {
						target,
						type,
						onClick = noop,
					} = props;

					const newProps = {
						...props,
						onClick: flow( onClick, fillProps.onClose ),
						target: `${ pluginName }/${ target }`,
					};

					switch ( type ) {
						case 'sidebar':
							return <PluginSidebarMoreMenuItem { ...newProps } />;
					}
					return null;
				} }
			</Fill>
		) }
	</PluginContext.Consumer>
);

PluginMoreMenuItem.Slot = ( { fillProps } ) => (
	<Slot name={ SLOT_NAME } fillProps={ fillProps } />
);

export default PluginMoreMenuItem;
