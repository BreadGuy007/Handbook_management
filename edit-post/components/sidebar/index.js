/**
 * External dependencies
 */
import { connect } from 'react-redux';

/**
 * WordPress Dependencies
 */
import { __ } from '@wordpress/i18n';
import { withFocusReturn } from '@wordpress/components';

/**
 * Internal Dependencies
 */
import './style.scss';
import PostSettings from './post-settings';
import BlockInspectorPanel from './block-inspector-panel';
import Header from './header';

import { getActivePanel } from '../../store/selectors';

const Sidebar = ( { panel } ) => {
	return (
		<div
			className="edit-post-sidebar"
			role="region"
			aria-label={ __( 'Editor advanced settings' ) }
			tabIndex="-1"
		>
			<Header />
			{ panel === 'document' && <PostSettings /> }
			{ panel === 'block' && <BlockInspectorPanel /> }
		</div>
	);
};

export default connect(
	( state ) => {
		return {
			panel: getActivePanel( state ),
		};
	},
	undefined,
	undefined,
	{ storeKey: 'edit-post' }
)( withFocusReturn( Sidebar ) );
