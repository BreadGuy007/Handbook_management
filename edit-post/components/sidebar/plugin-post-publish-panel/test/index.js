/**
 * External dependencies
 */
import { mount } from 'enzyme';

/**
 * WordPress dependencies
 */
import { SlotFillProvider } from '@wordpress/components';

/**
 * Internal dependencies
 */
import PluginPostPublishPanel from '../';

describe( 'PluginPostPublishPanel', () => {
	test( 'renders fill properly', () => {
		const wrapper = mount(
			<SlotFillProvider>
				<PluginPostPublishPanel
					className="my-plugin-post-publish-panel"
					title="My panel title"
					initialOpen={ true }
				>
					My panel content
				</PluginPostPublishPanel>
				<PluginPostPublishPanel.Slot />
			</SlotFillProvider>
		);

		expect( wrapper.find( 'Slot' ).children() ).toMatchSnapshot();
	} );
} );
