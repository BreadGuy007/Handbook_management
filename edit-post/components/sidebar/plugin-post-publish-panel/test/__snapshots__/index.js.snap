// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginPostPublishPanel renders fill properly 1`] = `
<PanelBody
  className="my-plugin-post-publish-panel"
  initialOpen={true}
  key="1---0/.0"
  title="My panel title"
>
  <div
    className="components-panel__body my-plugin-post-publish-panel is-opened"
  >
    <h2
      className="components-panel__body-title"
    >
      <Button
        aria-expanded={true}
        className="components-panel__body-toggle"
        onClick={[Function]}
      >
        <button
          aria-expanded={true}
          className="components-button components-panel__body-toggle"
          onClick={[Function]}
          type="button"
        >
          <SVG
            className="components-panel__arrow"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              aria-hidden="true"
              className="components-panel__arrow"
              focusable="false"
              height="24px"
              role="img"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <G>
                <g>
                  <Path
                    d="M0,0h24v24H0V0z"
                    fill="none"
                  >
                    <path
                      d="M0,0h24v24H0V0z"
                      fill="none"
                    />
                  </Path>
                </g>
              </G>
              <G>
                <g>
                  <Path
                    d="M12,8l-6,6l1.41,1.41L12,10.83l4.59,4.58L18,14L12,8z"
                  >
                    <path
                      d="M12,8l-6,6l1.41,1.41L12,10.83l4.59,4.58L18,14L12,8z"
                    />
                  </Path>
                </g>
              </G>
            </svg>
          </SVG>
          My panel title
        </button>
      </Button>
    </h2>
    My panel content
  </div>
</PanelBody>
`;
